### PR TITLE
refactor: change _uuid to _id

### PR DESCRIPTION
**Pull request by Bento007**
- #2778

### Reviewers
**Functional:** @danieljhegeman @seve 

**Readability:** @nayib-jose-gloria 

---

## Changes
- modify variable with suffix _uuid to have the suffix _id for the backend APIs, frontend, and processing pipeline.

### CODE REVIEW

File: backend/config/curation-api.yml [hunk 1]
```diff
@@ -148,13 +148,13 @@ paths:
       description: >-
         Update a private Collection or a Revision. If any Collection metadata fields are included in the body of the
         request, these fields SHALL be updated for the specified Collection.
-      operationId: backend.corpora.lambdas.api.v1.collection.update_collection
+      operationId: backend.corpora.lambdas.api.v1.curation.collections.collection_id.actions.patch
```
Comment (Bento007) [inline backend/config/curation-api.yml:151]: had to move some functions around

File: backend/config/curation-api.yml [hunk 2]
```diff
@@ -303,21 +303,21 @@ paths:
...
+  /v1/collections/{collection_id}/datasets/upload-link:
     put:
       summary: Upload a Dataset by providing a link
       description: >-
         Upload a Dataset. The Dataset will be uploaded from the provided link. MUST include a **curator_tag** in the
         request body; if an existing Dataset has this tag, the existing Dataset SHALL be replaced, otherwise a new
         Dataset will be added. MAY include the **id** of an existing Dataset, in which case the existing Dataset
         SHALL be replaced.
-      operationId: backend.corpora.lambdas.api.v1.collection_uuid.upload.relink
+      operationId: backend.corpora.lambdas.api.v1.curation.collections.collection_id.datasets.upload_link.put
```
Comment (Bento007) [inline backend/config/curation-api.yml:314]: had to move some functions around

File: backend/corpora/lambdas/api/v1/curation/collections/collection_id/datasets/upload_link.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from flask import make_response
+
+from backend.corpora.lambdas.api.v1.collection_uuid.upload import upload_from_link
+
+
+def put(collection_id: str, body: dict, token_info: dict):
```
Comment (Bento007) [inline backend/corpora/lambdas/api/v1/curation/collections/collection_id/datasets/upload_link.py:6]: new function for uploading from a link using the curator api.

File: tests/unit/backend/corpora/api_server/collection_uuid/test_create_genesets.py [hunk 4]
```diff
@@ -97,8 +97,8 @@ def test_geneset_creation__not_private_403(self):
         self.assertEqual(403, response.status_code)
 
     def test_geneset_creation__no_collection_403(self):
-        uuid = generate_uuid()
+        _id = generate_id()
```
Comment (Bento007) [inline tests/unit/backend/corpora/api_server/collection_uuid/test_create_genesets.py:100]: changed to _id since `id` is a python key word

**Review by Bento007 — COMMENTED** [review]
Highlighting changes that are beyond a rename

File: backend/corpora/dataset_submissions/app.py [hunk 5]
```diff
@@ -44,30 +44,30 @@ def dataset_submissions_handler(s3_event: dict, unused_context) -> None:
         logger.debug(parsed)
```
Comment (danieljhegeman) [inline backend/corpora/dataset_submissions/app.py:44]: @Bento007  Above on ln 41, and below on lns 89 and 112: is there any reason to not change the comments/logging from "uuid" to "id" as well?

**Review by nayib-jose-gloria — COMMENTED** [review]
Need rename a couple of directories--I caught `backend/corpora/lambdas/api/v1/collection_uuid` and `tests/unit/backend/corpora/api_server/collection_uuid`

**Review by seve — APPROVED** [review]

**Review by danieljhegeman — APPROVED** [review]
Looks good from a cursory inspection of instances of `uuid` in the code at this point.

Consider my comment about leaving "uuid" in a few places (comments, logs). I think it makes sense to migrate those to "id" as well though definitely not critical.

**Pull request merged**